blog/_posts/2021-02-16-improving-scala-library-ecosystem.md

```diff
@@ -0,0 +1,127 @@
+---
+layout: blog-detail
+post-type: blog
+by: Eugene Yokota, Julien Richard-Foy
+title: Improving the Scala Library Ecosystem
```

Comment:
I think the title should be more focused on the topic. "Improving the Scala Library Ecosystem" could be something completely different.

Reply:
"Improving the Scala Library Ecosystem with versionScheme"?

Reply:
Or maybe something that mentions "conflicting transitive dependencies"?

Reply:
“Evicting conflcting transitive dependencies”?

Reply:
Silent eviction and JAR-swapping is already happening. The improvement is that we make substitution of binary imcompatible JARs an error state by injecting version scheme information either in POM or user's `build.sbt`.

"Enforcing Semantic Versioning with versionScheme"?

Reply:
That's not bad, but it puts the emphasis on something negative-sounding (enforcement) rather than something positive. "Preventing version conflicts with versionScheme"?

Reply:
"In dependency hell? Here, have a nice cold glass of water" 😜

Reply:
I'm ok with either of Seth's suggestions.

Reply:
relevant parenting advice from Runar: https://twitter.com/runarorama/status/1358943798005481480

Reply:
My motivation for the initial loftier title is that unlike eviction warning where end users just suffer, the act of few (library authors) will going forward determine the strictness / looseness of the eviction linting.